nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/fetcher/EsFetcher.java

```diff
@@ -44,6 +50,18 @@ public DocumentModelListImpl fetchDocuments() {
         for (SearchHit hit : getResponse().getHits()) {
             // TODO: this does not work on multi repo
             doc = DocumentModelReaders.fromSource(hit.getSource()).sid(sid).getDocumentModel();
+            // Add highlight if it exists
+            Map<String, HighlightField> esHighlights = hit.highlightFields();
+            if (!esHighlights.isEmpty()) {
+                Map<String, List<String>> fields = new HashMap<>();
+                for (String field : esHighlights.keySet()) {
+                    fields.put(field, new ArrayList<>());
+                    for (Text fragment : esHighlights.get(field).getFragments()) {
```

Comment:
Instead of `esHighlights.keySet()` you can loop on `esHighlights.entrySet()` then use `field = entry.getKey()` and instead of `esHighlights.get(field)` use `entry.getValue()`.